### DISCUSSION TITLE
Scanning subdirectories

### DISCUSSION THREAD

**Discussion by HiImJulien** [rome/tools#4699] [Suggestions]
Hey there!

I am working on a monorepo, that contain things including the backend (written in 🦀) and the frontend (written in TS+React).

In order to keep everything tidy, I need to move files into the directories they belong. For instance, currently, there is only a single project/subdirectory that uses JS. This would mean, that the file `rome.json` belongs into that directory rather than staying in the workspace root.

**Request:** 

Would you consider to make the plugin recursively search for configuration files, rather than assuming that everything is in the root?

If a recursive search is not applicable due to some restrictions imposed by VSCode, would it at least be possible to specify "project" locations? (And retain the previous default behavior, if the property is either `undefined` or empty `[]`).

I.e.:

```json5
// settings.json or vscode workspace
{
    "rome.projects": [
      "${workspaceRoot}/project_a",
      "${workspaceRoot}/project_b"
    ]
}
```

**Comment by ematipico** [comment]
Rome already scans for `rome.json` files in parent directories. You can install rome in some subproject/sub-directory, and then have the config file at the root of your repo. That will work. That's how our repository is setup:
- That's where we have a script https://github.com/rome/tools/blob/main/website/package.json#L8
- That's where the config sits https://github.com/rome/tools/blob/main/rome.json

**Reply by HiImJulien** [reply]
> have the config file at the root of your repo

That is exactly what I am trying to avoid. Is there any way around it (without having to open two VSCode instances)?

EDIT:

I hope that there is no misunderstanding. The issue I am having now, is that if the `rome.json` is NOT in the root, then VSCode plugin does NOT pick it up and format my files.

But also, I do not want to have `rome.json` in the root of the monorepo but within a subdirectory that contains the actual project it is used for.

**Reply by ematipico** [reply]
Can I ask you to draw an exact proposal of what the feature would look like? I understand you have a specific use case, but I am stupid and I can't understand yet what you're looking for. 

Maybe an example would be great, because I feel that you have a particular use case but I fail to see it, sorry about that. Remember that CLI and LSP need to work both.

To clarify, I have never talked about the "root" of the project. Rome picks ups the first `rome.json` that it finds when looking traversing the parent directories.
